lib/vector-core/src/event/util/log/all_fields.rs

```diff
@@ -131,8 +131,14 @@ impl<'a> Iterator for FieldsIter<'a> {
                         }
                     }
                 },
-                Some(LeafIter::Root(value)) => {
-                    return Some(("message".to_owned(), value));
+                Some(LeafIter::Root((value, visited))) => {
+                    let result = if *visited {
+                        None
+                    } else {
+                        Some(("message".to_owned(), *value))
+                    };
```

Comment:
I recommend using `bool::then`:
```suggestion
                    let result = (!*visited).then(|| ("message".to_owned(), *value));
```